### PR TITLE
add city_and_region field

### PR DESCRIPTION
**Pull request by danpollock**
Hey @mark-walle : How does this look as a solution to the "Victoria, Texas" problem. 

I've left City as a dimension as well, but this one should do us better, I think, as a filter.

### CODE REVIEW

File: shared_fields_common.view.lkml [hunk 1]
```diff
@@ -114,6 +114,15 @@ view: shared_fields_common {
     group_label: "Location"
   }
 
+  dimension: city_and_region {
```
Comment (mark-walle) [inline shared_fields_common.view.lkml]: `geo_city_and_region` would match all other dimensions in the Location section.

File: shared_fields_common.view.lkml [hunk 2]
```diff
@@ -114,6 +114,15 @@ view: shared_fields_common {
...
+    type: string
+    sql: CASE
+       WHEN (${TABLE}.geo_country = 'CA') THEN ${TABLE}.geo_city || ', ' || ${TABLE}.geo_region_name
```
Comment (mark-walle) [inline shared_fields_common.view.lkml]: sometimes there is no value in `geo_city`, so there are a lot of "(blank), COUNTY" type results when filtering on the field.

File: shared_fields_common.view.lkml [hunk 3]
```diff
@@ -114,6 +114,15 @@ view: shared_fields_common {
...
+       ELSE ${TABLE}.geo_city || ', ' || ${TABLE}.geo_country
```
Comment (mark-walle) [inline shared_fields_common.view.lkml]: sometimes there is no value in geo_city, so there are a lot of "(blank), COUNTY" type results when filtering on the field.

File: shared_fields_common.view.lkml (hunk 1 above)
Comment (danpollock) [reply shared_fields_common.view.lkml]: Changed it, though I think down the road we likely want to remove the "geo_" from all of them.

File: shared_fields_common.view.lkml (hunk 3 above)
Comment (danpollock) [reply shared_fields_common.view.lkml]: Duplicated comment?

Comment (mark-walle) [reply shared_fields_common.view.lkml]: It was on the wrong line.

File: shared_fields_common.view.lkml (hunk 1 above)
Comment (mark-walle) [reply shared_fields_common.view.lkml]: We could also use the label parameter on these to change how they appear in the UI.

**Comment by danpollock** [comment]
Take a look now @mark-walle ?

**Comment by mark-walle** [comment]
That's working well.

**Comment by danpollock** [comment]
Thanks for the improvements suggested!

**Pull request merged**